raiden/transfer/channel.py

```diff
@@ -2787,6 +2786,14 @@ def state_transition(
             block_number=block_number,
             block_hash=block_hash,
         )
+    elif type(state_change) == ActionChannelCoopSettle:
```

Comment:
Shouldn't we rather use `is` for such type comparisons?

Reply:
That would probably be better.
But since those type of checks are all over the place, I would rather not have this refactored in this PR, and even less so only checking with `is` for the new code path.

Reply:
This would be a good candidate to also refactor with `functools.singledispatch` (c.f. #7252) (in another PR)